frontend/Makefile

```diff
@@ -0,0 +1,22 @@
+all: check-coverage lint build
+check-coverage:  test
+	CI=true yarn cover
+lint: format
+	yarn lint
+build: node_modules
+	yarn build
+test: node_modules
+	CI=true yarn test
+check-format: node_modules
+	yarn check-format
+format: node_modules
+	yarn format
+cover: node_modules
+	yarn cover
+start: node_modules
+	yarn start
+install: node_modules
+node_modules: package.json
+	yarn
```

Comment:
add the command `touch $@` after yarn to ensure that the `node_modules` timestamp is updated. It is possible that yarn installs something in a subdirectory of `node_modules` and the top-level directory does not get an updated timestamp. then it is always out-of-date with respect to `packages.json` and it will keep running `yarn`.

`$@` is the make special variable for the current target.

Reply:
Thanks for the detailed explanation.
Done.